metadata/src/main/java/org/apache/kafka/controller/QuorumFeatures.java

```diff
@@ -128,4 +128,27 @@ VersionRange localSupportedFeature(String featureName) {
     boolean isControllerId(int nodeId) {
         return quorumNodeIds.contains(nodeId);
     }
+
+    // check if all controller nodes are ZK Migration ready
+    public boolean isAllControllersZkMigrationReady() {
```

Comment:
Can we follow the same pattern that we've got in `reasonNotSupported` where we return an optional string of a reason why the quorum isn't ready? That way we can have all the pertinent logging come from KRaftMigrationDriver